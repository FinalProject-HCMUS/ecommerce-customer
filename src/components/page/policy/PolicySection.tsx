import type React from 'react';
import { motion } from 'framer-motion';
import type { PolicySectionType } from '../../../interfaces/temp/policy';

interface PolicySectionProps {
  section: PolicySectionType;
  index: number;
}

const PolicySection: React.FC<PolicySectionProps> = ({ section, index }) => {
  return (
    <motion.div
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      transition={{ delay: 0.2 + index * 0.1 }}
      className="mb-6"
    >
      <h2 className="text-xl font-semibold mb-2">{section.title}</h2>
      <p className="text-gray-700">{section.content}</p>
    </motion.div>
  );
};

export default PolicySection;
