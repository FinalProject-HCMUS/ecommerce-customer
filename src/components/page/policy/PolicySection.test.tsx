import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PolicySection from './PolicySection';

// Mock framer-motion to avoid animation-related issues in tests
jest.mock('framer-motion', () => ({
  motion: {
    div: ({ children, className }: { children: React.ReactNode, className: string }) => (
      <div data-testid="motion-div" className={className}>
        {children}
      </div>
    ),
  },
}));

// Mock the i18n translation function
jest.mock('../../../helpers/i18n', () => ({
  t: jest.fn((key: string) => {
    // Mock translation responses based on the key
    const translations: Record<string, string> = {
      'policy.0.title': 'Returns Policy',
      'policy.0.content': 'Our returns policy allows returns within 30 days of purchase.',
      'policy.1.title': 'Privacy Policy',
      'policy.1.content': 'We respect your privacy and protect your data.',
      'policy.2.title': 'Terms of Service',
      'policy.2.content': 'By using our service, you agree to these terms.',
    };
    return translations[key] || key;
  }),
}));

describe('PolicySection', () => {
  test('renders policy section with index 0', () => {
    render(<PolicySection index={0} />);
    
    // Check that the title and content are rendered correctly
    expect(screen.getByText('Returns Policy')).toBeInTheDocument();
    expect(screen.getByText('Our returns policy allows returns within 30 days of purchase.')).toBeInTheDocument();
    
    // Check that the container has the correct class
    const container = screen.getByTestId('motion-div');
    expect(container).toHaveClass('mb-6');
  });
  
  test('renders policy section with index 1', () => {
    render(<PolicySection index={1} />);
    
    // Check that the title and content are rendered correctly
    expect(screen.getByText('Privacy Policy')).toBeInTheDocument();
    expect(screen.getByText('We respect your privacy and protect your data.')).toBeInTheDocument();
  });
  
  test('renders policy section with index 2', () => {
    render(<PolicySection index={2} />);
    
    // Check that the title and content are rendered correctly
    expect(screen.getByText('Terms of Service')).toBeInTheDocument();
    expect(screen.getByText('By using our service, you agree to these terms.')).toBeInTheDocument();
  });
  
  test('renders policy section with non-existent index', () => {
    render(<PolicySection index={999} />);
    
    // When translation keys don't exist, the keys themselves should be rendered
    expect(screen.getByText('policy.999.title')).toBeInTheDocument();
    expect(screen.getByText('policy.999.content')).toBeInTheDocument();
  });
  
  test('applies correct heading styles', () => {
    render(<PolicySection index={0} />);
    
    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toHaveClass('text-xl', 'font-semibold', 'mb-2');
  });
  
  test('applies correct paragraph styles', () => {
    render(<PolicySection index={0} />);
    
    const paragraph = screen.getByText('Our returns policy allows returns within 30 days of purchase.');
    expect(paragraph).toHaveClass('text-gray-700');
  });
  
  test('matches snapshot', () => {
    const { container } = render(<PolicySection index={0} />);
    expect(container).toMatchSnapshot();
  });
});