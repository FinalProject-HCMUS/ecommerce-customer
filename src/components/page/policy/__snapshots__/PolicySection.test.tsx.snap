// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicySection matches snapshot 1`] = `
<div>
  <div
    class="mb-6"
    data-testid="motion-div"
  >
    <h2
      class="text-xl font-semibold mb-2"
    >
      Returns Policy
    </h2>
    <p
      class="text-gray-700"
    >
      Our returns policy allows returns within 30 days of purchase.
    </p>
  </div>
</div>
`;
